mdal/frmts/mdal_esri_tin.cpp

```diff
@@ -0,0 +1,179 @@
+/*
+ MDAL - Mesh Data Abstraction Library (MIT License)
+ Copyright (C) 2019 Vincent Cloarec (vcloarec at reos dot site)
+*/
+
+#include "mdal_esri_tin.hpp"
+
+//from https://stackoverflow.com/questions/41012414/convert-double-value-from-little-endian-to-big-endian
+template <typename T>
+static bool readAndChangeEndianness( T &value, std::ifstream &in )
+{
+  char *const p = reinterpret_cast<char *>( &value );
+  in.read( p, sizeof( T ) );
+  if ( in.eof() )
+    return false;
+
+  for ( size_t i = 0; i < sizeof( T ) / 2; ++i )
+    std::swap( p[i], p[sizeof( T ) - i - 1] );
+
+  return true;
+}
+MDAL::DriverEsriTin::DriverEsriTin(): Driver( "ESRI_TIN",
+                                              "Esri TIN",
+                                              "*.adf",
+                                              Capability::ReadMesh
+                                              )
+{}
+
+MDAL::Driver *MDAL::DriverEsriTin::create()
+{
+    return new DriverEsriTin();
+}
+
+std::unique_ptr<MDAL::Mesh> MDAL::DriverEsriTin::load( const std::string &uri, MDAL_Status *status )
+{
+    Vertices vertices;
+    Faces faces;
+
+    populateVertices( uri, vertices );
+    populateFaces( uri, faces );
+
+    std::unique_ptr< MemoryMesh > mesh(
+    new MemoryMesh(
+      name(),
+      vertices.size(),
+      faces.size(),
+      3,
+      computeExtent( vertices ),
+      uri
+    )
+  );
+
+  addTerrainElevationDatasetGroup( mesh.get(), vertices );
+
+  return mesh;
+}
+
+bool MDAL::DriverEsriTin::canRead(const std::string &uri)
+{
+
+    std::string zFileName = zFile( uri );
+    std::string faceFileName = faceFile( uri );
+
+    std::ifstream xyIn( xyFile( uri ), std::ifstream::in | std::ifstream::binary );
+    if ( ! xyIn.is_open() )
+        return false;
+
+    std::ifstream zIn( zFile( uri ), std::ifstream::in | std::ifstream::binary );
+    if ( ! zIn.is_open() )
+        return false;
+
+    std::ifstream faceIn( faceFile( uri ), std::ifstream::in | std::ifstream::binary );
+    if ( ! faceIn.is_open() )
+        return false;
+
+    //check if the file sizes of the nodes are coherent ?
+
+
+    return true;
+}
+
+std::string MDAL::DriverEsriTin::xyFile(const std::string &uri) const
+{
+    return pathJoin( dirName( uri ), "tnxy.adf" );
+}
+
+std::string MDAL::DriverEsriTin::zFile(const std::string &uri) const
+{
+    return pathJoin( dirName( uri ), "tnz.adf" );
+}
+
+std::string MDAL::DriverEsriTin::faceFile(const std::string &uri) const
+{
+    return pathJoin( dirName( uri ), "tnod.adf" );
+}
+
+void MDAL::DriverEsriTin::populateVertices( const std::string &uri, MDAL::Vertices &vertices )
+{
+    //firt, read the (x,y) coordinates
+    std::ifstream inXY( xyFile( uri ), std::ifstream::in | std::ifstream::binary );
+
+    if ( ! inXY.is_open() )
+        throw MDAL_Status::Err_FileNotFound;
+
+    inXY.ignore( 64 ); //ignore the 64 first bytes
```

Comment:
this is probably something to function canRead(). aren't those 64bytes some common string?

Reply:
No, see the other comment